## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#553a517b-c635-4db7-92e0-4b72060ad8bf) |
| mint-erasure.sh | :x: [more...](#586cd1dc-3d2c-4405-8431-d60532d6badf) |
| mint-dist-erasure.sh | :x: [more...](#7839d601-4754-46c0-b7d9-4dc5ac90ea14) |
| mint-zoned.sh | :x: [more...](#c27bf906-35f3-4d38-83d8-0850a32a6a78) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#d24bb330-3623-4a76-b6f5-90106d1a02dc) |
<a name="c27bf906-35f3-4d38-83d8-0850a32a6a78"></a>
### 12942-fcbe7d6/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s8:31677
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 45 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 55 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 38 seconds
(8/15) Running minio-dotnet tests ... done in 54 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 1 seconds
{
  "args": {
    "bucketName": "minio-go-test-q2ivtmubo21a5pzl",
    "objectName": "llos9hg6gsavj6eqajdmplskvpu4ki",
    "sse": {}
  },
  "duration": 576,
  "error": "unexpected EOF",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "Test \n, error: unexpected EOF",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 1 minutes and 33 seconds
(10/15) Running minio-js tests ... done in 51 seconds
(11/15) Running minio-py tests ... done in 2 minutes and 2 seconds
(12/15) Running s3cmd tests ... done in 27 seconds
(13/15) Running s3select tests ... done in 5 seconds
(14/15) Running versioning tests ... done in 3 minutes and 7 seconds

Executed 14 out of 15 tests successfully.
```
<a name="7839d601-4754-46c0-b7d9-4dc5ac90ea14"></a>
### 12942-fcbe7d6/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s7:30087
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 1 minutes and 51 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 25 seconds
(8/15) Running minio-dotnet tests ... done in 53 seconds
(9/15) Running minio-go tests ... FAILED in 55 seconds
{
  "args": {
    "bucketName": "minio-go-test-ih9rxmk0xbaz6gbp",
    "objectName": "2256bafz9yfkvmx6hby2ir4urrykvm",
    "sse": {}
  },
  "duration": 463,
  "error": "unexpected EOF",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "Test \n, error: unexpected EOF",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 53 seconds
(10/15) Running minio-js tests ... done in 49 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 29 seconds
(12/15) Running s3cmd tests ... done in 28 seconds
(13/15) Running s3select tests ... done in 4 seconds
(14/15) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 14 out of 15 tests successfully.
```
<a name="586cd1dc-3d2c-4405-8431-d60532d6badf"></a>
### 12942-fcbe7d6/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:31265
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 1 minutes and 49 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 23 seconds
(8/15) Running minio-dotnet tests ... done in 46 seconds
(9/15) Running minio-go tests ... FAILED in 44 seconds
{
  "args": {
    "bucketName": "minio-go-test-zzz4ihngcft1q1nu",
    "objectName": "olm3nlrw3s6kma1apiue0jywb5xm9j",
    "sse": {}
  },
  "duration": 4539,
  "error": "We encountered an internal error, please try again.: cause(io: read/write on closed pipe)",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 22 seconds
(10/15) Running minio-js tests ... done in 48 seconds
(11/15) Running minio-py tests ... done in 51 seconds
(12/15) Running s3cmd tests ... done in 23 seconds
(13/15) Running s3select tests ... done in 2 seconds
(14/15) Running versioning tests ... done in 3 minutes and 2 seconds

Executed 14 out of 15 tests successfully.
```
<a name="553a517b-c635-4db7-92e0-4b72060ad8bf"></a>
### 12942-fcbe7d6/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:31149
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 1 minutes and 50 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 34 seconds
(8/15) Running minio-dotnet tests ... done in 53 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 5 seconds
{
  "args": {
    "bucketName": "minio-go-test-6c9dormljjleuiuh",
    "objectName": "53poyos0vh6ad1go5mp0nfyj5tlosi",
    "sse": {}
  },
  "duration": 529,
  "error": "unexpected EOF",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "Test \n, error: unexpected EOF",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 1 minutes and 2 seconds
(10/15) Running minio-js tests ... done in 50 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 41 seconds
(12/15) Running s3cmd tests ... done in 27 seconds
(13/15) Running s3select tests ... done in 5 seconds
(14/15) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 14 out of 15 tests successfully.
```
<a name="d24bb330-3623-4a76-b6f5-90106d1a02dc"></a>
### 12942-fcbe7d6/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s4:30491
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 1 minutes and 52 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... FAILED in 9 seconds
{
  "name": "mc",
  "duration": "3285",
  "function": "test_get_object",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-20654 --quiet --no-color cp myminio/mc-test-bucket-23648/mc-test-object-29909 mc-test-object-29909.downloaded  >>> \n`myminio/mc-test-bucket-23648/mc-test-object-29909` -> `mc-test-object-29909.downloaded`\nmc: <ERROR> Failed to copy `http://minio-k8s4:30491/mc-test-bucket-23648/mc-test-object-29909`. Resource requested is unreadable, please reduce your request rate\nTotal: 0 B, Transferred: 0 B, Speed: 0 B/s"
}
(7/15) Running minio-dotnet tests ... done in 53 seconds
(8/15) Running minio-go tests ... FAILED in 1 seconds
{
  "args": {
    "bucketName": "minio-go-test-90e9lc9p03ijitpk",
    "fileName": "/tmp/FPutObjectTest669574760",
    "objectName": "minio-go-test-90e9lc9p03ijitpkFPutObject",
    "opts": ""
  },
  "duration": 1159,
  "error": "The specified key does not exist.",
  "function": "FPutObject(bucketName, objectName, fileName, opts)",
  "message": "FPutObject failed",
  "name": "minio-go: testFPutObjectV2",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 57 seconds
{
  "name": "minio-java",
  "function": "composeObject()",
  "duration": 47,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = NoSuchKey, message = The specified key does not exist., bucketName = minio-java-test-ojchmo, objectName = minio-java-test-vsj46e, resource = /minio-java-test-ojchmo/minio-java-test-vsj46e, requestId = 169A9627323C89A5, hostId = 69c7423a-6847-4b68-adac-e2986d04f20a)\nrequest={method=PUT, url=http://minio-k8s4:30491/minio-java-test-ojchmo/minio-java-test-vsj46e, headers=Content-Type: application/octet-stream\nHost: minio-k8s4:30491\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.3.0\nContent-MD5: y5LRepBMzsLm4juLtmJF+w==\nx-amz-content-sha256: 108eaaf9181151e84bcc41c5b1aa8246b76bde80282d81ae2f0c8caecb147232\nx-amz-date: 20210812T144150Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210812/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=404, headers=Accept-Ranges: bytes\nContent-Length: 367\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 169A9627323C89A5\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Thu, 12 Aug 2021 14:41:50 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:670), io.minio.S3Base.putObject(S3Base.java:1919), io.minio.S3Base.putObject(S3Base.java:1150), io.minio.MinioClient.putObject(MinioClient.java:1666), FunctionalTest.composeObject(FunctionalTest.java:2256), FunctionalTest.runObjectTests(FunctionalTest.java:3695), FunctionalTest.runTests(FunctionalTest.java:3720), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(8/15) Running minio-js tests ... FAILED in 30 minutes and 18 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 11,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:79:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:665:10) at endReadableNT (internal/streams/readable.js:1317:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(8/15) Running minio-py tests ... FAILED in 10 minutes and 19 seconds
{
  "name": "minio-py:test_get_object",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-d4edf3f6-8638-4118-9650-e08376fdec53",
    "object_name": "81326bcf-5c42-41a9-b608-d151f13841ec"
  },
  "message": "HTTPConnectionPool(host='minio-k8s4', port=30491): Max retries exceeded with url: /minio-py-test-d4edf3f6-8638-4118-9650-e08376fdec53/81326bcf-5c42-41a9-b608-d151f13841ec (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 913, in test_get_object\n    _test_get_object(log_entry, sse)\n  File \"/mint/run/core/minio-py/tests.py\", line 899, in _test_get_object\n    object_data = _CLIENT.get_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1165, in get_object\n    return self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 406, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 273, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 846, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 846, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 846, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s4', port=30491): Max retries exceeded with url: /minio-py-test-d4edf3f6-8638-4118-9650-e08376fdec53/81326bcf-5c42-41a9-b608-d151f13841ec (Caused by ResponseError('too many 503 error responses'))\n",
  "duration": 600908
}
(8/15) Running s3cmd tests ... FAILED in 7 seconds
{
  "name": "s3cmd",
  "duration": "711",
  "function": "test_get_object",
  "status": "FAIL",
  "error": "ERROR: S3 error: 503 (SlowDown): Resource requested is unreadable, please reduce your request rate"
}
(8/15) Running s3select tests ... done in 4 seconds
(9/15) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/xl-storage.go
+++ b/cmd/xl-storage.go
@@ -1074,7 +1074,7 @@ func (s *xlStorage) ReadVersion(ctx context.Context, volume, path, versionID str
 	} else {
 		buf, err = s.readMetadata(pathJoin(volumeDir, path, xlStorageFormatFile))
 		if err != nil {
-			if os.IsNotExist(err) {
+			if osIsNotExist(err) {
 				if err = Access(volumeDir); err != nil && osIsNotExist(err) {
 					return fi, errVolumeNotFound
 				}

